### PR TITLE
add dependabot.yml for C3 framework clis updates

### LINKED ISSUE #3805: 🚀 Feature Request: Use dependabot to update framework dependencies in C3
### Describe the solution

Let's use dependabot to keep the framework workers in C3 up to date.

In https://github.com/cloudflare/workers-sdk/pull/3776 we moved all the framework version info into a "fake" package.json so that we can rely on the existing dependabot automation to do all the hard stuff of keeping versions up to date and running e2e tests before PRs with version updates are merged.

The ideal outcome of this is that we have a PR created by dependabot for the workers-sdk repo any time a new fw version we declared in the package.json is released.

There are several things we'll need to think through here, for example:

- the workers-sdk used to use dependabot previously, but I believe it's currently disabled. @lrapoport-cf should have more info. So while we don't want to force wrangler or other projects in the monorepo to use dependabot, we should find a way to enable it just for c3, which seems to be possible (by specifying the [`directory` in the dependabot.yml](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directory)
- initially all PRs created by the bot should require human approval, we might revisit this in the future depending on our confidence in the automation
- I propose that we initially set the update interval to daily, just so that we can test the system more rapidly, but eventually move to weekly to reduce the noise.

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
resolves #3805 

____

Based on [my testing](https://github.com/dario-piotrowicz/dependabot-testing/pulls) this should work as expected, but I think we can only be 100% sure of it after we merge and check how it behaves.

### CODE REVIEW

**Comment by dario-piotrowicz** [comment]
Question, as you can see [here](https://github.com/dario-piotrowicz/dependabot-testing/pulls) with this we get one PR per dependency, although this is nice since it gives us more flexibility/granularity it could be annoying if we end up with too many smaller PRs

I should be able to group all the PRs together via the [groups config](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups)

But that could be more problematic/more cumbersome to deal with, for example in the case C3 has an issue with some but not all upgrades (we'd have to manually create/separate different PR for different updates)

Thoughts?

**Comment by IgorMinar** [comment]
@dario-piotrowicz as discussed, during the initial testing period (1-2 weeks) let's have it send individual PRs and based on how often they "just work" and can be merged without hassle we'll switch to weekly update cadence and potentially group all updates into a single PR. We need more data to make that decision, so let's get that first.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: 2
+updates:
+  # Automatically check for updates in framework CLIs for C3
+  - package-ecosystem: "npm"
+    directory: "/packages/create-cloudflare/src/frameworks"
+    schedule:
+      interval: "daily"
```
Comment (IgorMinar) [inline .github/dependabot.yml:7]: can you please add a comment that this is temporary for us to test the system over the next week or two, and then we intend to switch over to weekly?

**Review by IgorMinar — APPROVED** [review]
the rest looks good. this seems a bit "too easy" :-)

File: .github/dependabot.yml (hunk 1 above)
Comment (dario-piotrowicz) [reply .github/dependabot.yml:11]: comment added 🙂

**Pull request merged**